src/hooks/abilities/utils/getDeleteAbilities.ts

```diff
@@ -0,0 +1,185 @@
+import { TFunction } from 'i18next'
+import { P, match } from 'ts-pattern'
+
+import type { useBasicName } from '@app/hooks/useBasicName'
+import { checkSubname } from '@app/utils/utils'
+
+type BasicName = ReturnType<typeof useBasicName>
+
+type Result = {
+  canDelete: boolean
+  canDeleteContract?: 'nameWrapper' | 'registry'
+  canDeleteRequiresWrap?: boolean
+  canDeleteMethod?: 'setRecord' | 'setSubnodeOwner'
+  isPCCBurned?: boolean
+  isParentOwner?: boolean
+  canDeleteError?: string
+}
+
+const BASE_ABILITIES: Result = {
+  canDelete: false,
+}
+
+export const getDeleteAbilities = ({
+  name,
+  address,
+  basicNameData,
+  parentBasicNameData,
+  hasSubnames,
+  t,
+}: {
+  name: string
+  address?: string
+  basicNameData: BasicName
+  parentBasicNameData: BasicName
+  hasSubnames: boolean
+  t: TFunction
+}): Result => {
+  if (!checkSubname(name)) return BASE_ABILITIES
+  return (
+    match([basicNameData, parentBasicNameData])
+      /* --------------- WRAPPED WITH PCC BURNED --------------- */
+      /*
+       * WRAPPED NAME OWNER
+       * */
+      .with(
+        [
+          {
+            ownerData: {
+              ownershipLevel: 'nameWrapper',
+              owner: P.when((owner) => owner === address),
+            },
+            wrapperData: {
+              parent: {
+                PARENT_CANNOT_CONTROL: true,
+              },
+              child: P.select('fuses'),
+            },
+          },
+          {
+            ownerData: {
+              owner: P.select('parentOwner'),
+            },
+          },
+        ],
+        ({ fuses, parentOwner }) => ({
+          ...BASE_ABILITIES,
```

Comment:
what's the point in doing this? you overwrite the canDelete property in the line after